### PR TITLE
Proposed fix for build-time errors with (mp/fp)_s_rmap when debug ena…

### PR DESCRIPTION
**Pull request by kaleb-himes**
…bled and ecc disabled

### CODE REVIEW

File: wolfcrypt/src/tfm.c [hunk 1]
```diff
@@ -3076,12 +3076,9 @@ int mp_add_d(fp_int *a, fp_digit b, fp_int *c)
 #endif  /* HAVE_ECC || !NO_PWDBASED */
 
 
-#if defined(HAVE_ECC) || defined(WOLFSSL_KEY_GEN) || defined(HAVE_COMP_KEY)
```
Comment (SparkiDev) [inline wolfcrypt/src/tfm.c]: fp_s_rmap is only used if HAVE_ECC is defined.
Some compilers complain about unused static const data and others don't.

File: wolfcrypt/src/integer.c [hunk 2]
```diff
@@ -4729,12 +4729,9 @@ LBL_U:mp_clear (&v);
 #endif /* WOLFSSL_KEY_GEN */
 
 
-#if defined(HAVE_ECC) || defined(WOLFSSL_KEY_GEN) || defined(HAVE_COMP_KEY)
```
Comment (SparkiDev) [inline wolfcrypt/src/integer.c]: mp_s_rmap is only used if HAVE_ECC is defined.
Some compilers complain about unused static const data and others don't.

File: wolfcrypt/src/tfm.c (hunk 1 above)
Comment (kaleb-himes) [reply wolfcrypt/src/tfm.c]: Could you try running:

./configure --disable-ecc --enable-debug
./configure --disable-ecc --disable-fastmath --enable-debug

These are used when HAVE_ECC is not defined and debugging is enabled.

File: wolfcrypt/src/integer.c (hunk 2 above)
Comment (kaleb-himes) [reply wolfcrypt/src/integer.c]: Could you try running:

./configure --disable-ecc --enable-debug
./configure --disable-ecc --disable-fastmath --enable-debug

These are used when HAVE_ECC is not defined and debugging is enabled.

Comment (SparkiDev) [inline wolfcrypt/src/integer.c]: Some compilers complain about unused constant static variables.
I would change the list of conditionals to match those around both uses.

File: wolfcrypt/src/tfm.c (hunk 1 above)
Comment (SparkiDev) [inline wolfcrypt/src/tfm.c]: As above.

**Comment by kaleb-himes** [comment]
These are NOT unused compiler warnings. Please try the noted configurations?

```
./configure --disable-ecc --enable-debug && make
wolfcrypt/src/tfm.c: In function ‘mp_toradix’:
wolfcrypt/src/tfm.c:3346:18: error: ‘fp_s_rmap’ undeclared (first use in this function)
         *str++ = fp_s_rmap[d];
                  ^~~~~~~~~
wolfcrypt/src/tfm.c:3346:18: note: each undeclared identifier is reported only once for each function it appears in
  CC       src/src_libwolfssl_la-tls.lo
```

```
./configure --disable-ecc --enable-debug --disable-fastmath && make
src/.libs/libwolfssl.so: undefined reference to `mp_s_rmap'
collect2: error: ld returned 1 exit status
Makefile:2854: recipe for target 'wolfcrypt/test/testwolfcrypt' failed
make[1]: *** [wolfcrypt/test/testwolfcrypt] Error 1
make[1]: *** Waiting for unfinished jobs....
src/.libs/libwolfssl.so: undefined reference to `mp_s_rmap'
collect2: error: ld returned 1 exit status
src/.libs/libwolfssl.so: undefined reference to `mp_s_rmap'
collect2: error: ld returned 1 exit status
Makefile:2705: recipe for target 'examples/echoserver/echoserver' failed
make[1]: *** [examples/echoserver/echoserver] Error 1
src/.libs/libwolfssl.so: undefined reference to `mp_s_rmap'
collect2: error: ld returned 1 exit status
Makefile:2802: recipe for target 'tests/unit.test' failed
make[1]: *** [tests/unit.test] Error 1
Makefile:2835: recipe for target 'testsuite/testsuite.test' failed
make[1]: *** [testsuite/testsuite.test] Error 1
src/.libs/libwolfssl.so: undefined reference to `mp_s_rmap'
collect2: error: ld returned 1 exit status
Makefile:2848: recipe for target 'wolfcrypt/benchmark/benchmark' failed
make[1]: *** [wolfcrypt/benchmark/benchmark] Error 1
src/.libs/libwolfssl.so: undefined reference to `mp_s_rmap'
collect2: error: ld returned 1 exit status
Makefile:2679: recipe for target 'examples/client/client' failed
make[1]: *** [examples/client/client] Error 1
src/.libs/libwolfssl.so: undefined reference to `mp_s_rmap'
collect2: error: ld returned 1 exit status
Makefile:2692: recipe for target 'examples/echoclient/echoclient' failed
make[1]: *** [examples/echoclient/echoclient] Error 1
src/.libs/libwolfssl.so: undefined reference to `mp_s_rmap'
collect2: error: ld returned 1 exit status
Makefile:2749: recipe for target 'examples/server/server' failed
make[1]: *** [examples/server/server] Error 1
make[1]: Leaving directory '/home/boz-xeon2/wolfssl'
Makefile:2228: recipe for target 'all' failed
make: *** [all] Error 2

```

**Comment by kaleb-himes** [comment]
Being caught in DEBUG mode with this logic in tfm.c (similar in integer.c) which have no dependence on HAVE_ECC for mp_toradix

```
3254 #if defined(WOLFSSL_KEY_GEN) || defined(HAVE_COMP_KEY) || \                      
3255     defined(WOLFSSL_DEBUG_MATH) || defined(DEBUG_WOLFSSL) 
```

As I noted this may not be the optimal solution. If you have a more optimal solution I am open to suggestions!

**Comment by SparkiDev** [comment]
I got it wrong in the first set of comments - I didn't look for all references to the global.
You need to check all the defines so that the global variable is available in all circumstances.

**Comment by kaleb-himes** [comment]
Sean,

That makes perfect sense, don't want to compile unnecessary code in order to reduce footprint. Great consideration. Working on a more optimal solution.

**Pull request merged**